overlord/aspectstate/aspectstate.go

```diff
@@ -39,16 +39,30 @@ func SetAspect(databag aspects.DataBag, account, bundleName, aspect, field strin
 
 	asp := aspectBundle.Aspect(aspect)
 	if asp == nil {
-		return &aspects.AspectNotFoundError{Account: account, BundleName: bundleName, Aspect: aspect}
+		err := &aspects.NotFoundError{Cause: "aspect not found"}
+		return fillNotFoundErr(err, account, bundleName, aspect, field)
```

Comment:
I have never seen this pattern before to fill an incoming errors with more info. It's a bit odd because expectation of what field would be filled becomes fuzzy. Shouldn't we instead ensure that the level below has enough information to create a full error? or otherwise have separate richer errors here, though I think the former would be preferable

Reply:
Fixed this in the suggested way